src/resources/2023/CursedMonkeyPaw.ts

```diff
@@ -0,0 +1,118 @@
+import {
+  Item,
+  Location,
+  canAdventure,
+  getMonsters,
+  itemDropsArray,
+  Effect,
+  toEffect,
+  monkeyPaw,
+  prepareForAdventure,
+} from "kolmafia";
+import { have as have_ } from "../../lib";
+import { get } from "../../property";
+import { $item } from "../../template-string";
+import { clamp } from "../../utils";
+
+export const item = $item`cursed monkey's paw`;
+
+/**
+ * @returns Whether or not we currently `have` the cursed monkey's paw
+ */
+export function have(): boolean {
+  return have_(item);
+}
+
+/**
+ * @returns The number of monkey paw wishes we have remaining
+ */
+export function wishes(): number {
+  return clamp(5 - get("_monkeyPawWishesUsed"), 0, 5);
+}
+
+/**
+ * @returns A set of all items we expect to be able to wish; this doesn't actually constitute all items
+ */
+export function wishableItems(): Set<Item> {
+  return new Set(
+    ...Location.all()
+      .filter((l) => canAdventure(l))
+      .map((l) =>
+        getMonsters(l)
+          .filter((m) => m.copyable)
+          .map((m) =>
+            itemDropsArray(m)
+              .filter(
+                ({ type, rate, drop }) =>
+                  !drop.quest && (type !== "c" || rate >= 1) // Remove random roll drops
+              )
+              .map(({ drop }) => drop)
+          )
+      )
+      .flat()
+  );
+}
+let _unwishableEffects: Effect[];
+function unwishableEffects(): Effect[] {
```

Comment:
Can you add an explanatory comment here? This code isn't super readable to me at least

Reply:
done! took the opportunity to streamline it now that it's not a quick js one-liner